### DISCUSSION TITLE
Blank page when printing

### DISCUSSION THREAD

**Discussion by issck** [stephanrauh/ngx-extended-pdf-viewer#2576] [General]
Hi Stephan,

Thank you for the work and answers last time.

My app has a list of pdfs, they can be opened with the pdfViewer in a modal.
All pdfs can be opened correctly, but only the pdf first opened can be printed. The pdfs opened afterwards, when printed just show blank page.
I followed the debugging steps from the Troubleshooting, so only the first opened pdf has content in <printContainer>, the ones opened afterwards will have something like this 
`<div _ngcontent-ng-c39577623="" id="printContainer"></div>`

Do you have any ideas why that happened, or suggestion for further checking?

Thanks,
Colin

**Comment by stephanrauh** [comment]
No, I've never seen something like this before. Either printing works, or it doesn't. But I've never heard anybody complain about being able to print once, but not with a document opened later. I've tried your scenario on the showcase. It works as expected.

I'm afraid you've run into a brain twister. :(

**Reply by issck** [reply]
We display the pdf as BLOBs.
Then maybe a question is how do we generate content for printContainer?

**Reply by issck** [reply]
An update.

By removing #printContainer after modal close solves the issue.

**Reply by stephanrauh** [reply]
Sorry for answering late - it's because I'm ill. That said, which version are you using? I faintly remember there was a similar (or the same?) bug, and I remember solving it. But I'm not sure, and of course, I can't tell whether you're using the version containing the bug fix.

**Reply by issck** [reply]
Hope you are getting better.
I'm using 21.4.1

**Reply by stephanrauh** [reply]
I've got an idea. I sort of managed to reproduce your issue by deleting the printContainer manually. Unfortunately, my reproducer doesn't match your error description, so I'm not sure I've reproduced your bug. But I think it's better to create a print container when starting to print, and to remove after printing. 

That's necessary to support displaying two instances of the PDF viewer side-by-side. And as a side effect, it might solve your issue.

**Reply by stephanrauh** [reply]
@issck Please update to version 22.0.0-alpha.0. It should solve your issue.

That version is an alpha version because the pdf.js team has changed several ids, and I'm not sure I've already managed to fix every bug caused by renaming. Version 22.0.0 is going to update to pdf.js 4.7. The current alpha version aims at supporting both version 4.6 and 4.7, but that's likely to cause errors. Even so, please run a test with the alpha version. If you don't want to use it in production, that makes sense to me, but I guess version 22.0.0 is merely one or two weeks away, so running a test makes sense.

**Comment by JackmanBR** [comment]
Hi stephanrauh, 

I use extended and the display works perfectly, now I have two problems. The first is that when I click on the print button, blank pages appear in the print preview, if my PDF has 3 pages, for each page with content, the next page appears without blank content, for example, if my PDF has 3 pages , it looks like this (1-ok,2-blank,3-ok,4-blank,5-ok,6-blank), I use the current version 2.0.0-alpha.5. The other problem is that when viewing the PDF, the locale is wrong, that is, it should read pt-br, but the characters are swapped, for example, "R$" is swapped for "B" or "Â". What would be the solutions especially for the problem of blank pages? remembering that any pdf gives the problem of alternate blank pages, even upload pdf.
![image](https://github.com/user-attachments/assets/5282ebe4-7b53-4fb1-abd2-4e9b6056f61a)
![imagem_2024-10-16_185548005](https://github.com/user-attachments/assets/574a63a9-8d42-4a54-bb11-3698e8b69987)

**Comment by stephanrauh** [comment]
Sounds like a CSS problem to me. The print algorithm renders a canvas for every page, uses CSS to ensure the canvas covers the entire page, puts them into a div called "#printContainer", uses CSS to hide everything else and prints the `#printContainer`. Many CSS frameworks interfere with the CSS of ngx-extended-pdf-viewer. In your case, the pages seem to be slightly larger than one print page, so every second page is empty.

As for the character swapping: check your network tab. Maybe the PDF viewer tries to load a font that's not there?

Oh, and I'm confused about your version number. The most current stable version is 21.4.5. There's also a 21.4.6, which should also work fine if you're using the stable branch of pdf.js. "2.0.0-alpha.5" sounds like a typo, but even so, `*.alpha.5` is not a current version.
